org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/BackendUtils.java

```diff
@@ -1073,4 +1078,75 @@ public static void deriveWPD(TEntityType ci, List<String> errors) {
 		}
 		return res;
 	}
+
+	/**
+	 * Check the id's for new nodes, avoid repeating of target labels
+	 * @param duplicatedNode
+	 * @param tNodeTemplate
+	 * @param targetLabel
+	 */
+	public static TNodeTemplate checkId(TNodeTemplate duplicatedNode, TNodeTemplate tNodeTemplate, String targetLabel) {
+		if (tNodeTemplate.getId().contains(targetLabel)) {
+			duplicatedNode.setId(Util.makeNCName(tNodeTemplate.getId() + idCounterRequirements));
+			duplicatedNode.setName(Util.makeNCName(tNodeTemplate.getName() + idCounterRequirements));
+			idCounterRequirements++;
+			return duplicatedNode;
+		} else {
+			duplicatedNode.setId(Util.makeNCName(tNodeTemplate.getId() + "-" + targetLabel));
+			duplicatedNode.setName(Util.makeNCName(tNodeTemplate.getName() + "-" + targetLabel));
+			return duplicatedNode;
+		}
+	}
+
+	/**
+	 * Fix the Ids to avoid duplicate Ids in capabilities
+	 * @param nodeTemplate
+	 * @return fixedCapabilities with fixed Ids
+	 */
+	public static TNodeTemplate.Capabilities cloneCapabilities(TNodeTemplate nodeTemplate) {
+		if (nodeTemplate.getCapabilities() == null) {
+			LOGGER.info("Capabilites are null, not cloned any capabilities for " +  nodeTemplate.getId());
```

Comment:
This should be DEBUG. It is good that you use the logger, but it should use the right log level. This is not a user information , but more a DEBUG information, isn't it?

DEBUG is also logged as default (currently) - https://github.com/eclipse/winery/blob/master/org.eclipse.winery.repository/src/main/resources/logback.xml